### ISSUE TITLE
Order confirmation is sent before payment has been processed

### ISSUE THREAD

**Issue by Skwecode** [Nets-eCom/Magento2_easy#36]
Hi,

When you click, like in this screenshot, on "betal" or "pay", an order is created instantly and an order confirmation is sent to both the store owner & the customer. In my opinion, the email should be postponed until the order is actually completed / processing. I see that there's a "SubmitObserverPlugin.php", but it doesn't look like it's actually preventing any email from going out. This is happening when the status is still "Pending payment" & not "processing". We pack orders based on order confirmation emails, but this bug is forcing us to actually verify the payment before we can process the package, which shouldn't be needed. This doesn't happen when it's a card payment. This is on the latest version of the module, Magento version 2.4.2 & PHP7.4
![Screenshot 2022-07-07 at 15 00 46](https://user-images.githubusercontent.com/70141130/177779785-9d312b03-89c5-41d7-89ab-eb59d2ca1004.png) 

![Screenshot 2022-07-07 at 15 16 25](https://user-images.githubusercontent.com/70141130/177782808-c0c9f4f1-67ea-44d0-8c2f-17bf3f11c939.png)

![Screenshot 2022-07-07 at 15 16 01](https://user-images.githubusercontent.com/70141130/177782709-3d91ce08-eea6-4675-9ec5-055741bee4a4.png)

 EDIT:
After testing a bit more, even if you write faulty payment details using a card payment, the order confirmation is sent & an order is placed. "SubmitObserverPlugin.php" is never called.

**Comment by Skwecode** [comment]
Sorry for bugging you @Tim-Badin & @sursi, but have you seen this issue?

**Comment by Tim-Badin** [comment]
Thanks for ping @Skwecode ! We will investigate.

**Comment by Skwecode** [comment]
Hi @Tim-Badin, 
Do you have update? Are you able to replicate the issue?

**Comment by Tim-Badin** [comment]
Hi @Skwecode 

Our team has not been able to reproduce the issue. 
Please could you write to integration-plugin@nets.eu so that we can check the logs and your environment, sharing
- FTP access
- Temporary admin access

Thanks
